api/src/main/java/marquez/db/JobVersionDao.java

```diff
@@ -206,15 +210,17 @@ default void upsertOutputDatasetFor(UUID jobVersionUuid, UUID outputDatasetUuid)
    * @param jobVersionUuid The unique ID of the job version.
    * @param datasetUuid The unique ID of the output dataset
    * @param ioType The {@link IoType} of the dataset.
+   * @param jobUuid The unique ID of the job.
    */
   @SqlUpdate(
       """
     INSERT INTO job_versions_io_mapping (
-      job_version_uuid, dataset_uuid, io_type)
-    VALUES (:jobVersionUuid, :datasetUuid, :ioType)
+      job_version_uuid, dataset_uuid, io_type, job_uuid)
+    VALUES (:jobVersionUuid, :datasetUuid, :ioType, :jobUuid)
     ON CONFLICT DO NOTHING
```

Comment:
There's a unique index on:

```
job_version_uuid, dataset_uuid, io_type
```

We'll need to update the index to also include `job_uuid`

Reply:
I was able to test inserting  static lineage `job` event in `job_versions_io_mapping`, but I had to make the following changes to the table:

```sql
# (1) To allow for job_version_uuid = NULL, we have to remove the foreign key constraint
ALTER TABLE job_versions_io_mapping DROP CONSTRAINT job_versions_io_mapping_job_version_uuid_fkey;
ALTER TABLE job_versions_io_mapping ALTER COLUMN job_version_uuid DROP NOT NULL;

# (2) To add job_uuid to the unique constraint, we first drop the primary key, then recreate it; note given that job_version_uuid can be NULL, we need to check that job_version_uuid != NULL before inserting (duplicate columns otherwise)
ALTER TABLE job_versions_io_mapping DROP CONSTRAINT job_versions_io_mapping_pkey;
ALTER TABLE job_versions_io_mapping ADD CONSTRAINT job_versions_io_mapping_pkey UNIQUE (job_version_uuid,dataset_uuid,io_type,job_uuid);

# (3) Simulate static lineage job event (no job version)
INSERT INTO job_versions_io_mapping (job_version_uuid, dataset_uuid, io_type, job_uuid)
  VALUES (NULL, '4abaa4d6-f874-4d99-a7a4-d0791352b62c', 'INPUT', '86f988b5-3a94-4ded-9a19-fd2a062b0c3a');
 
# (4) Simulate run-level event (with job version)
INSERT INTO job_versions_io_mapping (job_version_uuid, dataset_uuid, io_type, job_uuid)
  VALUES ('c5894e50-c459-4667-a862-3e9aa3051dfa', '4abaa4d6-f874-4d99-a7a4-d0791352b62c', 'INPUT', '86f988b5-3a94-4ded-9a19-fd2a062b0c3a');
```

Reply:
The uniqueness of `job_version_uuid, dataset_uuid, io_type`, again, deserves a test which I am happy to implement. 

I think we should not accept null values for `job_version`. It would not make sense for table called `job_versions_io_mapping`.  I think we can leave in that case `job_versions_io_mapping_pkey` as is.